src/frontend/js/utils/test/factories.ts

```diff
@@ -140,6 +141,40 @@ export const CourseFactory = createSpec({
   orders: null,
 });
 
+export const RichieFullCourseFactory = createSpec({
```

Comment:
Richie is not meaningful here I think. If your problematic is not distinguish Joanie Course from Richie Course resources, maybe you could use `IndexedCourseFactory` as from Richie Course are fetched from elasticsearch. WDYT?
```suggestion
export const IndexedCourseFactory = createSpec({
```